src/App.js

```diff
@@ -3,10 +3,20 @@ import React, { useEffect, useState } from 'react';
 import TodoList from './TodoList'
 import AddTodoForm from './AddTodoForm'
 
+function useSemiPersistentState () {
+  const [todoList, setTodoList] = useState (JSON.parse(localStorage.getItem("savedTodoList")) || [])
+
+  useEffect (() => {
+    localStorage.setItem("savedTodoList", JSON.stringify(todoList))
+  }, [todoList])
+
+  return [todoList, setTodoList]
+}
+
 function App() {
+  const [todoList, setTodoList] = useSemiPersistentState()
+
 
-  const [todoList, setTodoList] = useState([])
-  
   function addTodo (newTodo) {
     setTodoList([...todoList, newTodo])
   }
```

Comment:
So this is my fault -- After more research, I found the way we set the new array was wrong. React allows us to provide a function to the setter (`setTodoList`) and function param will get the most recent value of the state.

So it'd look something like this:
```suggestion
    setTodoList( (previousTodoList) => [...previousTodoList, newTodo])
  }
```
I used an arrow function there b/c it's easier to read. How does all this sound to you?

Reply:
Exercise 1.4 overview section actually explains my comment a bit better 😆 
https://github.com/Code-the-Dream-School/react-cockatoo/wiki/Lesson-1.4
take a look